src/core/handle.ml

```diff
@@ -4,20 +4,33 @@ open Extra
 open Timed
 open Console
 open Terms
-open Print
 open Sign
 open Pos
 open Files
 open Syntax
+open Sig_state
 open Scope
+open Print
 
-(* Register a check for the type of builtin successor symbol ["+1"]. *)
+(* Register a check for the type of the builtin symbol "0" and "+1". *)
 let _ =
-  let expected_successor pos map =
-    let typ_0 = lift !((Builtin.get pos map "0").sym_type) in
+  let register = Builtin.register_expected_type (Unif.eq []) pp_term in
+  let expected_zero_type ss _pos =
+    try
+      match !((StrMap.find "+1" ss.builtins).sym_type) with
+      | Prod(a,_) -> a
+      | _ -> assert false
+    with Not_found -> Meta (fresh_meta Type 0, [||])
+  in
+  register "0" expected_zero_type;
+  let expected_succ_type ss _pos =
+    let typ_0 =
+      try lift !((StrMap.find "0" ss.builtins).sym_type)
+      with Not_found -> _Meta (fresh_meta Type 0) [||]
+    in
     Bindlib.unbox (_Impl typ_0 typ_0)
   in
-  Builtin.Check.register_expected_type "+1" expected_successor
+  register "+1" expected_succ_type
```

Comment:
What was wrong with the previous approach? This seemed like a good fit for this pattern.

Reply:
It was depending on the order the builtins where defined. See added examples. LP was failing if +1 was defined before 0.